src/core/core.scale.js

```diff
@@ -1,5 +1,8 @@
 'use strict';
 
+var moment = require('moment');
+moment = typeof(moment) === 'function' ? moment : window.moment;
```

Comment:
Agree with @etimberg's [comment](/chartjs/Chart.js/pull/4268/files#r117124932), Moment should not leak in `core.scale.js` but stays confined in `scale.time.js`